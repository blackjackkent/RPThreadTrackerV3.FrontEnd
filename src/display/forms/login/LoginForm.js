import React from 'react';
import PropTypes from 'prop-types';
import { AvField } from 'availity-reactstrap-validation';
import validator from '../../views/login/_loginFormValidator';

const propTypes = {
	handleInputChange: PropTypes.func.isRequired
};

const LoginForm = (props) => {
	const { onUsernameChange } = props;
	return (
		
	);
};

LoginForm.propTypes = propTypes;
export default LoginForm;
