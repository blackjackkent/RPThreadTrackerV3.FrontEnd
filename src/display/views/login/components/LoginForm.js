import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Row, Col, CardBody, Button, Label } from 'reactstrap';
import { AvForm, AvField } from 'availity-reactstrap-validation';
import LoadingIndicator from '~/display/shared/loading/LoadingIndicator';
import Card from '~/display/shared/styled/Card';
import validator from '../_loginFormValidator';

const propTypes = {
	isLoading: PropTypes.bool.isRequired,
	errorMessage: PropTypes.string.isRequired,
	onInputChange: PropTypes.func.isRequired,
	onSubmit: PropTypes.func.isRequired
};

const LoginForm = ({ isLoading, errorMessage, onInputChange, onSubmit }) => {
	return (
		<Card className="login-box p-4">
			<CardBody className="card-body">
				{isLoading && (
					<LoadingIndicator
						style={{
							width: 50,
							height: 50,
							position: 'absolute',
							top: 0,
							right: 0
						}}
					/>
				)}
				<AvForm onValidSubmit={onSubmit}>
					<h1>Login</h1>
					<p className="text-muted">Sign in to RPThreadTracker</p>
					{errorMessage && (
						<div className="has-danger">
							<p className="form-control-feedback">{errorMessage}</p>
						</div>
					)}
					<div>
						<div>
							<Label for="username">Username</Label>
							<AvField
								name="username"
								placeholder="Username"
								type="text"
								onChange={onInputChange}
								validate={validator.username}
							/>
						</div>
						<div>
							<Label for="password">Password</Label>
							<AvField
								name="password"
								placeholder="Password"
								type="password"
								onChange={onInputChange}
								validate={validator.password}
							/>
						</div>
					</div>
					<Row>
						<Col xs="6">
							<Button color="primary" role="button" className="px-4">
								Login
							</Button>
						</Col>
						<Col xs="6" className="text-right text-muted">
							<Link href="/register" to="/register">
								Sign up
							</Link>{' '}
							&bull;{' '}
							<Link href="/forgotpassword" to="/forgotpassword">
								Forgot password?
							</Link>
						</Col>
					</Row>
				</AvForm>
			</CardBody>
		</Card>
	);
};

LoginForm.propTypes = propTypes;
export default LoginForm;
