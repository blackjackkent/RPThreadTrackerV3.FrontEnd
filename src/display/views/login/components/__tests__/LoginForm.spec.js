// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import { getSpecWrapper } from '~/testhelpers/helpers.unit';
import LoginForm from '../LoginForm';
// #endregion imports

const createTestProps = (propOverrides) => ({
	isLoading: false,
	errorMessage: '',
	onInputChange: jest.fn(),
	onSubmit: jest.fn(),
	...propOverrides
});

describe('rendering', () => {
	describe('snapshots', () => {
		it('should render valid snapshot', () => {
			const props = createTestProps();
			const jsx = <LoginForm {...props} />;
			const element = shallow(jsx);
			expect(element).toMatchSnapshot();
		});
	});
});

describe('behavior', () => {
	describe('validation', () => {
		it('should validate the username field', () => {
			const props = createTestProps();
			const jsx = <LoginForm {...props} />;
			const element = shallow(jsx);
			const field = getSpecWrapper(element, 'username-field').find('AvField');
			expect(field.props().validate.required).toHaveProperty('value', true);
			expect(field.props().validate.minLength).toHaveProperty('value', 3);
			expect(field.props().validate.maxLength).toHaveProperty('value', 256);
		});
		it('should validate the password field', () => {
			const props = createTestProps();
			const jsx = <LoginForm {...props} />;
			const element = shallow(jsx);
			const field = getSpecWrapper(element, 'password-field').find('AvField');
			expect(field.props().validate.required).toHaveProperty('value', true);
		});
	});
	describe('onInputChange', () => {
		it('should be called when username changes', () => {
			const onInputChange = jest.fn();
			const props = createTestProps({
				onInputChange
			});
			const jsx = <LoginForm {...props} />;
			const element = shallow(jsx);
			const field = getSpecWrapper(element, 'username-field').find('AvField');
			field.simulate('change');
			expect(onInputChange).toHaveBeenCalledTimes(1);
		});
		it('should be called when password changes', () => {
			const onInputChange = jest.fn();
			const props = createTestProps({
				onInputChange
			});
			const jsx = <LoginForm {...props} />;
			const element = shallow(jsx);
			const field = getSpecWrapper(element, 'password-field').find('AvField');
			field.simulate('change');
			expect(onInputChange).toHaveBeenCalledTimes(1);
		});
	});
});
