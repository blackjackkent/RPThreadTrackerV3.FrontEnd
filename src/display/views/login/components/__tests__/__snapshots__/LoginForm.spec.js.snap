// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<Styled(Card)
  className="login-box p-4"
>
  <CardBody
    className="card-body"
    tag="div"
  >
    <AvForm
      data-spec="login-form-container"
      method="get"
      model={Object {}}
      onInvalidSubmit={[Function]}
      onKeyDown={[Function]}
      onSubmit={[Function]}
      onValidSubmit={[MockFunction]}
      tag={[Function]}
      validationEvent={
        Array [
          "onChange",
          "onInput",
        ]
      }
    >
      <h1>
        Login
      </h1>
      <p
        className="text-muted"
      >
        Sign in to RPThreadTracker
      </p>
      <div>
        <div
          data-spec="username-field"
        >
          <AvField
            name="username"
            onChange={[MockFunction]}
            placeholder="Username"
            type="text"
            validate={
              Object {
                "maxLength": Object {
                  "errorMessage": "Your username is too long.",
                  "value": 256,
                },
                "minLength": Object {
                  "errorMessage": "Your username must be more than 3 characters.",
                  "value": 3,
                },
                "required": Object {
                  "errorMessage": "You must enter a username.",
                  "value": true,
                },
              }
            }
          />
        </div>
        <div
          data-spec="password-field"
        >
          <AvField
            name="password"
            onChange={[MockFunction]}
            placeholder="Password"
            type="password"
            validate={
              Object {
                "required": Object {
                  "errorMessage": "You must enter a password.",
                  "value": true,
                },
              }
            }
          />
        </div>
      </div>
      <Row
        tag="div"
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="6"
        >
          <Button
            className="px-4"
            color="primary"
            tag="button"
          >
            Login
          </Button>
        </Col>
        <Col
          className="text-right text-muted"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="6"
        >
          <Link
            href="/register"
            replace={false}
            to="/register"
          >
            Sign up
          </Link>
           
          â€¢
           
          <Link
            href="/forgotpassword"
            replace={false}
            to="/forgotpassword"
          >
            Forgot password?
          </Link>
        </Col>
      </Row>
    </AvForm>
  </CardBody>
</Styled(Card)>
`;
